core/Tracker/Visit.php

```diff
@@ -553,11 +553,16 @@ private function getVisitStandardLength()
      */
     private function setIdVisitorForExistingVisit($valuesToUpdate)
     {
-        // Might update the idvisitor when it was forced or overwritten for this visit
         if (strlen($this->visitProperties->getProperty('idvisitor')) == Tracker::LENGTH_BINARY_ID) {
             $valuesToUpdate['idvisitor'] = $this->visitProperties->getProperty('idvisitor');
         }
-
+        
+        $visitorId = $this->request->getVisitorId();
+        if ($visitorId && strlen($visitorId) === Tracker::LENGTH_BINARY_ID) {
```

Comment:
this will ensure when cookies are enabled to then update the visitorId based on the value which was generated in the JS cookie.

while cookies were disabled it should have used some randomly generated visitorId generated by the backend.

We can't really think of any breaking issue because of this.